third_party/move/tools/move-package-cache/src/canonical.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use move_core_types::account_address::AccountAddress;
+use url::Url;
+
+/// Sanitizes a URL by replacing dots with underscores and converting to lowercase.
+///
+/// TODO: The current scheme is not bijective - if the original URL contains underscores, they will
```

Comment:
Just wonder whether this will lead to any issue? Have we considered using percent-encoding to avoid ambiguity?